src/firewall/core/rich.py

```diff
@@ -655,16 +670,25 @@ def check(self):
                 raise FirewallError(errors.INVALID_RULE, "forward-port and action")
 
         # source-port
-        elif type(self.element) == Rich_SourcePort:
-            if not functions.check_port(self.element.port):
-                raise FirewallError(errors.INVALID_PORT, self.element.port)
-            if self.element.protocol not in [ "tcp", "udp", "sctp", "dccp" ]:
+        elif type(self.element) == Rich_SourcePort:	
+            if not functions.check_port(self.element.port):	
+                raise FirewallError(errors.INVALID_PORT, self.element.port)	
+            if self.element.protocol not in [ "tcp", "udp", "sctp", "dccp" ]:	
                 raise FirewallError(errors.INVALID_PROTOCOL, self.element.protocol)
-
-        # other element and not empty?
-        elif self.element is not None:
-            raise FirewallError(errors.INVALID_RULE, "Unknown element %s" % 
-                                type(self.element))
+        
+        # tcp-mss-clamp
+        elif type(self.element) == Rich_Tcp_Mss_Clamp:
+            if self.element.value:
+                if (self.element.value).isdigit():
+                    if int(self.element.value) < 536:
+                        raise FirewallError(errors.INVALID_RULE, self.element.value)
+                elif self.element.value != "pmtu":
+                    raise FirewallError(errors.INVALID_RULE, self.element.value)
```

Comment:
This seems redundant with the checks in `checkTcpMssClamp()`. Why not just call that function?